File: IOS/2016-03-30 ObjrctC PageViewController/viewPager/ViewController.m
//
//  ViewController.m
//  viewPager
//
//  Created by woong on 2016. 3. 29..
//  Copyright © 2016년 handstudio. All rights reserved.
//

#import "ViewController.h"


@interface ViewController ()

@end

@implementation ViewController



- (void)viewDidLoad {
    [super viewDidLoad];

    //데이터 초기화
    [self initData];
}


/**
 * 데이터 초기화
 */
- (void) initData {

    categotyList = [NSArray arrayWithObjects: @"morning",@"Morning",@"Afternoon", nil];
    
    //PageViewController 초기화
    [self initPageViewController];
}


/**
 * PageViewController 초기화
 */
- (void) initPageViewController {
    
    //pageViewController 생성
    pageViewController = [[UIPageViewController alloc] initWithTransitionStyle:
                          UIPageViewControllerTransitionStyleScroll navigationOrientation:UIPageViewControllerNavigationOrientationHorizontal options:nil];
    pageViewController.dataSource = self;
    
    //pageViewController 에 첫번째 화면 초기화
    UIViewController *startingViewController = [self viewControllerAtIndex:0];
    NSArray *viewControllers = @[startingViewController];
    [pageViewController setViewControllers:viewControllers direction:UIPageViewControllerNavigationDirectionForward animated:NO completion:nil];
    
    //pageViewController 사이즈 조정
    pageViewController.view.frame = self.viewPager.bounds;
    
    //pageViewController 화면 viewController 초기화
    [self addChildViewController: pageViewController];
    [self.viewPager addSubview:pageViewController.view];
    [pageViewController didMoveToParentViewController:self];
    
}


/**
 * 해당 인덱스 추출
 */
- (NSUInteger)indexOfViewController:(PageViewController *)viewController {
    return [categotyList indexOfObject:viewController.name];
}


/**
 * 해당 인덱스의 화면 반환
 */
- (UIViewController *) viewControllerAtIndex:(NSUInteger)index {
    
    if (([categotyList count] == 0) || (index >= [categotyList count])) {
        return nil;
    }
    
    PageViewController *controller;
    
    if(index == 0) {
        controller = [[FirstViewController alloc] initWithNibName:@"FirstViewController" bundle:nil];
    } else if (index == 1) {
        controller = [[SecondViewController alloc] initWithNibName:@"SecondViewController" bundle:nil];
    } else {
        controller = [[ThirdViewController alloc] initWithNibName:@"ThirdViewController" bundle:nil];
    }
    
    controller.name = [categotyList objectAtIndex:index];
    
    return controller;
}


/**
 * 이전 페이지 로드
 */
- (UIViewController *)pageViewController:(UIPageViewController *)pageViewController viewControllerBeforeViewController:(UIViewController *)viewController {
    
    
    NSUInteger index = [self indexOfViewController:(PageViewController *)viewController];
    index--;
    
    return [self viewControllerAtIndex:index];
}


/**
 * 다음 페이지 로드
 */
- (UIViewController *)pageViewController:(UIPageViewController *)pageViewController viewControllerAfterViewController:(UIViewController *)viewController {
    
    NSUInteger index = [self indexOfViewController:(PageViewController *)viewController];
    index++;
    
    return [self viewControllerAtIndex:index];
}


@end

File: IOS/2016-03-29 pageSample/pageSample/ViewController.swift
//
//  ViewController.swift
//  pageSample
//
//  Created by woong on 2016. 3. 28..
//  Copyright © 2016년 handstudio. All rights reserved.
//

import UIKit

class ViewController: UIViewController , UIPageViewControllerDataSource {

    var pageViewController : UIPageViewController!
    var pageTitles : NSArray!
    var pageImages : NSArray!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        self.pageTitles = NSArray(objects: "첫번째 페이지" , "두번째 페이지", "세번째 페이지" , "네번째 페이지", "다섯번째 페이지")
        self.pageImages = NSArray(objects: "img01" , "img02" , "img03" , "img04" , "img05")
        
        self.pageViewController = self.storyboard?.instantiateViewControllerWithIdentifier("PageViewController") as! UIPageViewController
        
        self.pageViewController.dataSource = self
        
        let startVC = self.viewControllerAtIndex(0) as ContentViewController
        let viewControllers = NSArray(object: startVC)
        
        self.pageViewController.setViewControllers(viewControllers as? [UIViewController] , direction: .Forward, animated: true, completion: nil)
        self.pageViewController.view.frame = CGRectMake(0,30,self.view.frame.width, self.view.frame.height - 100)
        
        self.addChildViewController(self.pageViewController)
        self.view.addSubview(self.pageViewController.view)
        
    }
    
    
    /**
    * viewPageController 구성 함수
    */
    func viewControllerAtIndex (index : Int) -> ContentViewController {
        
        let vc : ContentViewController = self.storyboard?.instantiateViewControllerWithIdentifier("ContentViewController") as! ContentViewController
        
        vc.pageIndex = index
        vc.imageFile = self.pageImages[index] as! String
        vc.titleText = self.pageTitles[index] as! String
        
        print(">>> : " ,vc.titleText)
        
        return vc
    }
    
    
    
    /**
    * 이전 ViewPageController 구성
    */
    func pageViewController(pageViewController: UIPageViewController, viewControllerBeforeViewController viewController: UIViewController) -> UIViewController? {
        
        let vc = viewController as! ContentViewController
        var index = vc.pageIndex as Int

        if( index == 0 || index == NSNotFound) {
            return nil
        }
        
        index--
        
        return self.viewControllerAtIndex(index)
    }
    
    
    /**
    * 이후 ViewPageController 구성
    */
    func pageViewController(pageViewController: UIPageViewController, viewControllerAfterViewController viewController: UIViewController) -> UIViewController? {
        
        let vc = viewController as! ContentViewController
        
        var index = vc.pageIndex as Int
        
        if( index == NSNotFound) {
            return nil
        }
        
        index++
        
        if (index == self.pageTitles.count) {
            return nil
        }
        
        return self.viewControllerAtIndex(index)
    }
    
    
    /**
    * 인디케이터의 총 갯수
    */
    func presentationCountForPageViewController(pageViewController: UIPageViewController) -> Int {
        return self.pageTitles.count
    }
    
    
    /**
    * 인디케이터의 시작 포지션
    */
    func presentationIndexForPageViewController(pageViewController: UIPageViewController) -> Int {
        return 0
    }

}

